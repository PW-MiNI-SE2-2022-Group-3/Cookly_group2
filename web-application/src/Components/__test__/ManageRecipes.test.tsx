import React, { useState } from "react";
import ReactDOM from "react-dom";
import {
  render,
  screen,
  fireEvent,
  queryByTestId,
  cleanup,
} from "@testing-library/react";
import ManageRecipes from "../ManageRecipes";

it("renders correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ManageRecipes />, div);
});

it("renders buttons correctly", () => {
  const { getByTestId } = render(<ManageRecipes />);
  expect(getByTestId("filter-button")).toBeInTheDocument();
  expect(getByTestId("add-button")).toBeInTheDocument();
  expect(getByTestId("filter-button")).toHaveTextContent("Tag Filter");
  expect(getByTestId("add-button")).toHaveTextContent("Add Recipe");
});

it("renders input fields correctly", () => {
  const { getByTestId } = render(<ManageRecipes />);
  expect(getByTestId("search-textfield")).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = render(<ManageRecipes />);
  expect(tree).toMatchSnapshot();
});
