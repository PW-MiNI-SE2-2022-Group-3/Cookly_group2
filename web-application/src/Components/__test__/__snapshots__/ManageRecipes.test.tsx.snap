// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        style="width: 100%; margin: auto; padding-top: 20px;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-6u1dxy-MuiFormControl-root-MuiTextField-root"
          data-testid="search-textfield"
        >
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="outlined-basic"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-khze64-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-91aoiw-MuiButtonBase-root-MuiButton-root"
          data-testid="add-button"
          tabindex="0"
          type="button"
        >
          Add Recipe
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FoodBankIcon"
            focusable="false"
            style="margin-left: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3 4 9v12h16V9l-8-6zm.5 9.5c0 .83-.67 1.5-1.5 1.5v4h-1v-4c-.83 0-1.5-.67-1.5-1.5v-3h1v3h.5v-3h1v3h.5v-3h1v3zM15 18h-1v-3.5h-1v-3c0-1.1.9-2 2-2V18z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-91aoiw-MuiButtonBase-root-MuiButton-root"
          data-testid="filter-button"
          tabindex="0"
          type="button"
        >
          Tag Filter
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FilterListIcon"
            focusable="false"
            style="margin-left: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root tablecontainer css-11xur9t-MuiPaper-root-MuiTableContainer-root"
          style="border-radius: 0;"
        >
          <table
            class="MuiTable-root table css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: left; color: white;"
                >
                  ID
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: left; color: white;"
                >
                  NAME
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: left; color: white;"
                >
                  INSTRUCTIONS
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: center; color: white;"
                >
                  TAGS
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: center; color: white;"
                >
                  INGREDIENTS
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: center; color: white;"
                >
                  ACTIONS
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
            <tfoot
              class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeSmall MuiTablePagination-root css-m0bqz3-MuiTableCell-root-MuiTablePagination-root"
                  colspan="6"
                >
                  <div
                    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
                  >
                    <div
                      class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                    />
                    <p
                      class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                      id="mui-23"
                    >
                      Rows per page:
                    </p>
                    <div
                      class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                      variant="standard"
                    >
                      <select
                        aria-label="rows per page"
                        class="MuiTablePagination-select MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                        id="mui-22"
                      >
                        <option
                          class="MuiTablePagination-menuItem"
                          value="5"
                        >
                          5
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="10"
                        >
                          10
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="25"
                        >
                          25
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="-1"
                        >
                          All
                        </option>
                      </select>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                    <p
                      class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                    >
                      0–0 of 0
                    </p>
                    <div
                      class="MuiBox-root css-1zye22"
                    >
                      <button
                        aria-label="first page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="FirstPageIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="previous page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="KeyboardArrowLeftIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="next page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="KeyboardArrowRightIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="last page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="LastPageIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      style="width: 100%; margin: auto; padding-top: 20px;"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-6u1dxy-MuiFormControl-root-MuiTextField-root"
        data-testid="search-textfield"
      >
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="outlined-basic"
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-khze64-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RefreshIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-91aoiw-MuiButtonBase-root-MuiButton-root"
        data-testid="add-button"
        tabindex="0"
        type="button"
      >
        Add Recipe
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FoodBankIcon"
          focusable="false"
          style="margin-left: 10px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3 4 9v12h16V9l-8-6zm.5 9.5c0 .83-.67 1.5-1.5 1.5v4h-1v-4c-.83 0-1.5-.67-1.5-1.5v-3h1v3h.5v-3h1v3h.5v-3h1v3zM15 18h-1v-3.5h-1v-3c0-1.1.9-2 2-2V18z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-91aoiw-MuiButtonBase-root-MuiButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        Tag Filter
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FilterListIcon"
          focusable="false"
          style="margin-left: 10px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root tablecontainer css-11xur9t-MuiPaper-root-MuiTableContainer-root"
        style="border-radius: 0;"
      >
        <table
          class="MuiTable-root table css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: left; color: white;"
              >
                ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: left; color: white;"
              >
                NAME
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: left; color: white;"
              >
                INSTRUCTIONS
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: center; color: white;"
              >
                TAGS
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: center; color: white;"
              >
                INGREDIENTS
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: center; color: white;"
              >
                ACTIONS
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
          <tfoot
            class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeSmall MuiTablePagination-root css-m0bqz3-MuiTableCell-root-MuiTablePagination-root"
                colspan="6"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
                >
                  <div
                    class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                  />
                  <p
                    class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                    id="mui-23"
                  >
                    Rows per page:
                  </p>
                  <div
                    class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                    variant="standard"
                  >
                    <select
                      aria-label="rows per page"
                      class="MuiTablePagination-select MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                      id="mui-22"
                    >
                      <option
                        class="MuiTablePagination-menuItem"
                        value="5"
                      >
                        5
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="25"
                      >
                        25
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="-1"
                      >
                        All
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                  >
                    0–0 of 0
                  </p>
                  <div
                    class="MuiBox-root css-1zye22"
                  >
                    <button
                      aria-label="first page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="FirstPageIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="previous page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="next page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="last page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LastPageIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
